src/vmm/src/signal_handler.rs

```diff
@@ -229,23 +136,60 @@ impl SignalManager {
     }
 
     /// Handle the signal according to its signal number.
-    fn handle_signal(siginfo: signalfd_siginfo) {
-        match siginfo.ssi_signo as i32 {
-            SIGXFSZ => sigxfsz_handler(siginfo),
-            SIGXCPU => sigxcpu_handler(siginfo),
-            SIGBUS => sigbus_handler(siginfo),
-            SIGSEGV => sigsegv_handler(siginfo),
-            SIGPIPE => sigpipe_handler(siginfo),
-            SIGSYS => sigsys_handler(siginfo),
-            SIGHUP => sighup_handler(siginfo),
-            SIGILL => sigill_handler(siginfo),
-            // This should never be reached since we'll only receive the signals in the signalfd mask.
-            // Safe because we're terminating the process anyway.
+    fn handle_signal(info: signalfd_siginfo) {
+        let si_signo = info.ssi_signo as i32;
+        let si_code = info.ssi_code;
+
+        // For SIGSYS, we have some special logging.
+        if si_signo == SIGSYS {
+            log_sigsys_err(info);
+        }
+
+        // Increment the right metrics.
+        match si_signo {
```

Comment:
I would rather have a function that does this match and returns a tuple with the metric and exit code associated with each signal.

Reply:
done